### PR TITLE
Logistics: Add issue, pull request templates and code of conduct

### PR DESCRIPTION
**Pull request by yo1995**
## Description

It's recommended to have issue/pull request templates and a code of conduct for the repo: https://github.com/Esri/arcgis-runtime-samples-ios/community and I always wanted to standardize the format of PR description.

- [Issue templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository)
- [Instructions for adding pull request templates](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/creating-a-pull-request-template-for-your-repository)
- [Instructions for adding code of conduct](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-code-of-conduct-to-your-project#adding-a-code-of-conduct-manually)

Please feel free to...
- share your opinions on whether we should include these docs
- make changes to the wording and content

## Linked Issue(s)

- Please refer to `common-samples/issues/2317`

### CODE REVIEW

**Comment by yo1995** [comment]
This is the original text that I included in the PR template. I don't want to jam bunch of explanation in the comments, so I only keep the section headers and essential pieces.

```txt
<!--
	Examples for PR titles:
	- New: `Export vector tiles`
	- Update: `Create and save map` to use named user
	- Fix: Add alert for route tracker licensing error
	- Release 100.12.0.1
	- Logistics: Add pull request templates and code of conduct
-->


## Description
<!--
	At minimum, add one or two sentences describing the new/updated/fixed sample.
	If the sample has multiple features, use an unordered list.
	Including a link to the sample's directory on current branch is a plus!
	E.g. This PR implements `SAMPLE_NAME` in `SAMPLE_CATEGORY` category.
-->


## Linked Issue(s)
<!--
	Provide a truncated URL to the internal issue.
	E.g. `common-samples/issues/1000`, `cocoa/issues/10000`, etc. Issues, PRs in public GitHub repositories of other platforms can also be pasted here.
-->
- `common-samples/issues/`


<!-- 
## Screenshots
-->
<!--
	Optionally, include H.264 video, GIF, or screenshots to show the expected behaviors for a complex sample.
	If the PR updates an existing sample, consider providing before and after comparison screenshots.
	|Before|After|
	|:-:|:-:|
	|||
-->


## How To Test
<!--
	For simple samples, describe what is expected upon sample launch. 
	For complex samples, provide a numbered list of steps to get the expected result.
-->


<!-- 
## To Discuss
-->
<!--
	Optionally, if there are points that a reviewer needs to pay attention to, list them here.
-->
```

**Review by vquach2404 — DISMISSED** [review]
This is a great idea! I don't have any suggestions at the moment. I think we can improve as we use the templates more and find what we like/dislike.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
+---
+name: Bug Report
+about: Create a bug report to help us improve
+title: "[BUG] "
+labels: bug
+---
+
+**Describe the Bug**
+<!-- Provide an overview of the issue and why you consider it to be a bug -->
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/bug_report.md:9]:
```suggestion
**Describe the Bug**

<!-- Provide an overview of the issue and why you consider it to be a bug -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+<!-- How was the issue found? -->
+<!-- Add a video or screenshots if it helps -->
+
+**Expected behavior**
+<!-- What is the expected outcome? -->
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/bug_report.md:14]:
```suggestion
**Expected behavior**

<!-- What is the expected outcome? -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+**Steps To Reproduce**
+<!-- Clear concise steps to reproduce the issue -->
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/bug_report.md:17]:
```suggestion
**Steps To Reproduce**

<!-- Clear concise steps to reproduce the issue -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+<!-- Include code in collapsible section below -->
+
+1. Go to '...'
+2. Click on '....'
+3. Scroll down to '....'
+4. See error
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/bug_report.md:23]: Should these be in comments as well?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+**Build and Platform Information**
+<!-- Identify build and platform information -->
+
+* OS: [e.g. iOS 15.1, macOS 12.0]
+* Device: [e.g. iPhone 13, iPhone Simulator]
+* SDK Version: [e.g. 100.11.0]
+* App Release Version: [e.g. 100.11.0]
+* Which release it last worked (only if regression)?:
+* Commit(only if building the project):
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/bug_report.md:43]:
```suggestion
* Commit (only if building the project):
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+<details><summary>Reproducible Code</summary>
+
+```swift
+Swift repro code goes here
+```
+
+</details>
+
+**Build and Platform Information**
+<!-- Identify build and platform information -->
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/bug_report.md:36]:
```suggestion
**Build and Platform Information**

<!-- Identify build and platform information -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+* OS: [e.g. iOS 15.1, macOS 12.0]
+* Device: [e.g. iPhone 13, iPhone Simulator]
+* SDK Version: [e.g. 100.11.0]
+* App Release Version: [e.g. 100.11.0]
+* Which release it last worked (only if regression)?:
+* Commit(only if building the project):
+
+**Data and Credentials**
+<!-- Include any/all that are applicable -->
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/bug_report.md:46]:
```suggestion
**Data and Credentials**

<!-- Include any/all that are applicable -->
```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+**Additional Context**
+<!-- Add any other context about the problem here. -->
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/bug_report.md:49]:
```suggestion
**Additional Context**

<!-- Add any other context about the problem here. -->
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
+
+**Describe the solution you'd like**
+<!-- A clear and concise description of what you want to happen. -->
+
+**Describe alternatives you've considered**
+<!-- A clear and concise description of any alternative solutions or features you've considered. -->
+
+**Additional context**
+<!-- Add any other context or screenshots about the feature request here. -->
```
Comment (philium) [inline .github/ISSUE_TEMPLATE/feature_request.md:18]:
```suggestion
**Is your feature request related to a problem? Please describe.**

<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->

**Describe the solution you'd like**

<!-- A clear and concise description of what you want to happen. -->

**Describe alternatives you've considered**

<!-- A clear and concise description of any alternative solutions or features you've considered. -->

**Additional context**

<!-- Add any other context or screenshots about the feature request here. -->
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+
+## Linked Issue(s)
+
+- `common-samples/issues/`
+
+
+## How To Test
+
+
```
Comment (philium) [inline .github/PULL_REQUEST_TEMPLATE.md:15]:
```suggestion
## Description

<!--
	This PR implements `SAMPLE_NAME` in `SAMPLE_CATEGORY` category.
	Branch: URL_TO_BRANCH
-->

## Linked Issue(s)

<!-- - `common-samples/issues/` -->

## How To Test

```

**Review by philium — COMMENTED** [review]
Looks good! I left mostly just spacing suggestions.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 11]
```diff
@@ -0,0 +1,35 @@
+## Description
+
+<!--
+	This PR implements `SAMPLE_NAME` in `SAMPLE_CATEGORY` category.
```
Comment (philium) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: I just noticed that these are tabs, not spaces. Did you mean to do that? There isn't anything wrong with tabs per se, but I don't think we use them anywhere else. I just want to check that that was intentional.

Comment (yo1995) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: Good point. I should've set the text editor to `Indent use space`. Lemme redo the indentation.

**Review by philium — APPROVED** [review]

**Review by vquach2404 — APPROVED** [review]

**Comment by yo1995** [comment]
Thank you! 🥳 

image

**Pull request merged**